src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -355,14 +355,20 @@ private bool RunTests(TestRunCriteria testRunCriteria, ITestRunEventsRegistrar t
         /// Update Extensions path folder in testadapterspaths in runsettings.
         /// </summary>
         /// <param name="settingsXml"></param>
-        private static string UpdateExtensionsFolderInRunSettings(string settingsXml)
+        private static string UpdateExtensionsFolderInRunSettings(string settingsXml, List<string> sources)
         {
             if (string.IsNullOrWhiteSpace(settingsXml))
             {
                 return settingsXml;
             }
 
             var extensionsFolder = Path.Combine(Path.GetDirectoryName(typeof(TestPlatform).GetTypeInfo().Assembly.Location), "Extensions");
+            var sourcesPath = sources.Select(source => Path.GetDirectoryName(source)).ToList();
```

Comment:
Can we combine all the three operations? Like Select(...).Concat(extensionsFolder).Distinct().Join(";")